src/Common/src/Interop/Windows/secur32/Interop.SSPI.cs

```diff
@@ -0,0 +1,509 @@
+
+            // Connection semantics must be used.
+            Connection = 0x00000800,
+
+            // Client applications requiring extended error messages specify the
+            // ISC_REQ_EXTENDED_ERROR flag when calling the InitializeSecurityContext
+            // Server applications requiring extended error messages set
+            // the ASC_REQ_EXTENDED_ERROR flag when calling AcceptSecurityContext.
+            InitExtendedError = 0x00004000,
+            AcceptExtendedError = 0x00008000,
+            // A transport application requests stream semantics
+            // by setting the ISC_REQ_STREAM and ASC_REQ_STREAM
+            // flags in the calls to the InitializeSecurityContext
+            // and AcceptSecurityContext functions
+            InitStream = 0x00008000,
+            AcceptStream = 0x00010000,
+            // Buffer integrity can be verified; however, replayed
+            // and out-of-sequence messages will not be detected
+            InitIntegrity = 0x00010000,       // ISC_REQ_INTEGRITY
+            AcceptIntegrity = 0x00020000,       // ASC_REQ_INTEGRITY
+
+            InitManualCredValidation = 0x00080000,   // ISC_REQ_MANUAL_CRED_VALIDATION
+            InitUseSuppliedCreds = 0x00000080,   // ISC_REQ_USE_SUPPLIED_CREDS
+            InitIdentify = 0x00020000,   // ISC_REQ_IDENTIFY
+            AcceptIdentify = 0x00080000,   // ASC_REQ_IDENTIFY
+
+            ProxyBindings = 0x04000000,   // ASC_REQ_PROXY_BINDINGS
+            AllowMissingBindings = 0x10000000,   // ASC_REQ_ALLOW_MISSING_BINDINGS
+
+            UnverifiedTargetName = 0x20000000,   // ISC_REQ_UNVERIFIED_TARGET_NAME
+        }
+
+        internal enum Endianness
+        {
+            Network = 0x00,
+            Native = 0x10,
+        }
+
+        internal enum CredentialUse
+        {
+            Inbound = 0x1,
+            Outbound = 0x2,
+            Both = 0x3,
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct _CERT_CHAIN_ELEMENT
+        {
+            public uint cbSize;
+            public IntPtr pCertContext;
+            // Since this structure is allocated by unmanaged code, we can
+            // omit the fileds below since we don't need to access them
+            // CERT_TRUST_STATUS   TrustStatus;
+            // IntPtr                pRevocationInfo;
+            // IntPtr                pIssuanceUsage;
+            // IntPtr                pApplicationUsage;
+        }
+
+        // SecPkgContext_IssuerListInfoEx
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct IssuerListInfoEx
+        {
+            public SafeHandle aIssuers;
+            public uint cIssuers;
+
+            public unsafe IssuerListInfoEx(SafeHandle handle, byte[] nativeBuffer)
+            {
+                aIssuers = handle;
+                fixed (byte* voidPtr = nativeBuffer)
+                {
+                    // if this breaks on 64 bit, do the sizeof(IntPtr) trick
```

Comment:
The comment is from the original code. I'd keep it for now since it uncovers an issue in marshaling the struct from a native buffer: There is an assumption made that padding won't be added on any any architecture. 
Changed the comment to reflect this and linked to #3114.